econml/iv/dml/_dml.py

```diff
@@ -761,6 +761,14 @@ def predict(self, Y, T, X=None, W=None, Z=None, sample_weight=None, groups=None)
             TX_pred = np.tile(TX_pred.reshape(1, -1), (T.shape[0], 1))
         Y_res = Y - Y_pred.reshape(Y.shape)
         T_res = TXZ_pred.reshape(T.shape) - TX_pred.reshape(T.shape)
+        if T_res.sum() == 0:
```

Comment:
Again, the issue is not that the residuals sum to zero, it's that every entry is zero.

```suggestion
        if not T_res.any():
```